addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -196,15 +197,11 @@ const LinkDialog = Dialog.extend({
      * @override
      */
     save: function () {
-        this.linkWidget.save();
-        if (!this.linkWidget.final_data) {
-            // Invalid form content: do not proceed with save.
-            return;
-        }
-        this.final_data = this.linkWidget.final_data;
-        if (this.final_data) {
-            return this._super(...arguments);
-        }
+        const _super = this._super.bind(this);
+        return this.linkWidget.save().then(() => {
+            this.final_data = this.linkWidget.final_data;
+            return _super();
```

Comment:
`_super(...arguments)` always do that even if no argument otherwise it is subject to errors if we ever add an argument or a custom module wants to do it.